src/compiler/checker.ts

```diff
@@ -17081,7 +17081,7 @@ namespace ts {
             if (targetDeclarationKind !== SyntaxKind.Unknown) {
                 const decl = getDeclarationOfKind(resolvedRequire, targetDeclarationKind);
                 // function/variable declaration should be ambient
-                return isInAmbientContext(decl);
+                return decl.flags & NodeFlags.Ambient;

```

Comment:
`!!(decl.flags & NodeFlags.Ambient)`, otherwise the return type becomes `number | boolean`.